src/apispec/core.py

```
@@ -18,6 +18,55 @@
 VALID_METHODS = {2: VALID_METHODS_OPENAPI_V2, 3: VALID_METHODS_OPENAPI_V3}
 
 
+def get_ref(obj_type, obj, openapi_major_version):
+    """Return object or rererence
+
+    If obj is a dict, it is assumed to be a complete description and it is returned as is.
+    Otherwise, it is assumed to be a reference name as string and the corresponding $ref
+    string is returned.
+
+    :param str obj_type: "parameter" or "response"
+    :param dict|str obj: parameter or response in dict form or as ref_id string
+    :param int openapi_major_version: The major version of the OpenAPI standard
+    """
+    if isinstance(obj, dict):
+        return obj
+    return build_reference(obj_type, openapi_major_version, obj)
+
+
+def clean_parameters(parameters, openapi_major_version):
+    """Ensure that all parameters with "in" equal to "path" are also required
+    as required by the OpenAPI specification, as well as normalizing any
+    references to global parameters and checking for duplicates parameters
+
+    See https ://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#parameterObject.
+
+    :param list parameters: List of parameters mapping
+    :param int openapi_major_version: The major version of the OpenAPI standard
+    """
```

AFAIK, at this stage, a parameter can be either a string (reference) or a dictionary.

Since `name` and `in` are required in a parameter, shouldn't we assume that if it's a `dict`, it contains `name` and `in`?

Just wondering, I didn't check. But this would allow us to simplify this function.

(We could even raise a custom exception if `name` or `in` is missing, rather than let the code crash. That would make debugging easier. This is out of scope and could be done in another PR.)

---

You're right, OpenAPI requires `name` and `in` for a parameter object, so I've created a separate commit for this: d53eb9b but let me know if you prefer to move that commit into a new PR

---

Same PR is fine. I just didn't want you to feel you had to do it to get this PR merged. Thanks for doing it.